x-pack/plugins/search_playground/server/lib/fetch_query_source_fields.ts

```diff
@@ -141,22 +160,41 @@ const isFieldInIndex = (
 
 export const parseFieldsCapabilities = (
   fieldCapsResponse: FieldCapsResponse,
-  aggDocs: Array<{ index: string; doc: SearchResponse }>
+  aggMappingDocs: Array<{ index: string; doc: SearchResponse; mapping: IndicesGetMappingResponse }>
 ): IndicesQuerySourceFields => {
   const { fields, indices: indexOrIndices } = fieldCapsResponse;
   const indices = Array.isArray(indexOrIndices) ? indexOrIndices : [indexOrIndices];
 
-  const indexModelIdFields = aggDocs.map<IndexFieldModel>((aggDoc) => {
+  const indexModelIdFields = aggMappingDocs.map<IndexFieldModel>((aggDoc) => {
     const modelIdFields = Object.keys(aggDoc.doc.aggregations || {}).map<FieldModelId>((field) => {
       return {
         field,
         modelId: (aggDoc.doc.aggregations![field] as any)?.buckets?.[0]?.key,
       };
     });
 
+    const mappingProperties = aggDoc.mapping[aggDoc.index].mappings.properties || {};
```

Comment:
Maybe we can set up type here to avoid ts-ignore below?

Reply:
thanks - i added a temp interface for this mapping.